src/change/sdp.rs

```diff
@@ -1151,18 +1195,43 @@ impl From<MediaKind> for MediaType {
 }
 
 struct AsSdpParams<'a, 'b> {
-    pub candidates: &'a [Candidate],
-    pub creds: &'a IceCreds,
+    pub candidates: Cow<'a, [Candidate]>,
+    pub creds: IceCreds,
     pub fingerprint: &'a Fingerprint,
     pub setup: Setup,
     pub pending: Option<&'b Changes>,
 }
 
 impl<'a, 'b> AsSdpParams<'a, 'b> {
     pub fn new(rtc: &'a Rtc, pending: Option<&'b Changes>) -> Self {
+        // TODO: This might warrant its own function it's a bit much...
+        let (creds, candidates) = pending
+            .and_then(|p| {
+                p.iter().find_map(|c| match c {
+                    Change::IceRestart(creds) => {
+                        let new_candidates = Cow::Owned(
+                            rtc.ice
+                                .local_candidates()
+                                .iter()
+                                .map(|c| c.replace_creds(creds.ufrag.clone()))
+                                .collect(),
+                        );
+
+                        Some((creds.clone(), new_candidates))
+                    }
+                    _ => None,
+                })
+            })
+            .unwrap_or_else(|| {
+                (
+                    rtc.ice.local_credentials().clone(),
+                    Cow::Borrowed(rtc.ice.local_candidates()),
+                )
+            });
+
```

Comment:
Let's land #308 and do a simple for-loop here with the pending ice restart creds, if there is any.

Reply:
Let's use a bit less combinators here, it gets a bit too muddled what's going on.

We can use the new "SdpPendingOffer::ice_restart()` to make this

```rust
let (creds, candidates) = if let Some(creds) = p.and_then(|p| p.ice_restart()) {
...
} else { 
...
}
```

Reply:
Was just looking at this, we can't do that because `p` isn't `SdpPendingOffer` here, its' `&Changes` but I'll figure something out

Reply:
Oh. Hm… Let's move that ice_creds to `Changes`

`SdpPendingOffer` is after all just a facade against the API user.